docs/DeveloperGuide.md

```diff
@@ -8,30 +29,194 @@
+in the `BuildManager` class, passing `addBuild` the created build object with the provided name. If the provided name is 
+invalid, the `BlankStringException` exception will be thrown and an error message will be printed.
+
+The `addBuild()` method will check if the provided name is already in the list of builds. If the provided name is not
+in the list of builds, the method will add the build into the list of builds. If the provided name is already in the 
+list of builds, the `DuplicateBuildException` exception will be thrown and an error message will be printed.
+
+After adding the build into the list, the program will return to the `mainParseAdd` method where the method will print
+a message, telling the user the build has been added.
+
+Finally, the `mainParseAdd` method will call the `saveBuilds()` method in the `Storage` class to save the list of builds
+into the data file.
+
+The following sequence diagram shows how the add build operation works:
+![](/images/BuildManagerAddBuildSequence.png)
+
+### Edit Mode
+
+#### Build 
+
+![](/images/Build.png)
+
+A build is a collection of components that are used to create a PC. A build can be saved and loaded from the storage. 
+The user can also export the build to a text file via the export functionality. 
+
+In our application build is a class object which contains a 2D linked hash map of components. LinkedHashMap2D is a class
+representing a 2D linked hash map data structure. The keys of the outer map are the component types and the keys of the inner
+are the component names. The values of the inner map are the components themselves.
+
+The user can add, delete, and view the components in the build. The user can also
+perform aggregate operations on the build such as viewing the total price of the build and viewing the total wattage of the build.
+There are also check compatibility functions that check if the build's components are compatible with each other in different ways. Such as:
+- Check if the power supply is compatible with the build's total power consumption
+- Checking if the build's motherboard is compatible with the build's CPU and Cooler
+- Checking if the form factor of the build's case is compatible with the build's motherboard
+- Checking if the build's storage is compatible with the case expansion slots
+- Checking if the number of the build's GPUs is compatible with the motherboard GPU slots
+- Checking if the number of the build's RAM is compatible with the motherboard RAM slots
+
+##### Add & delete component
+
+The user can add and delete components from the build. When the user adds a component, the component is added to the 2D linked hash map. 
+When the user deletes a component, the component is removed from the 2D linked hash map. This is done by calling the `addComponent()` and `deleteComponent()` methods of the Build class.
+Which in turn calls the `addElement()` and `removeElement()` methods of the LinkedHashMap2D class.
+
+![](/images/BuildSequence.png)
```

Comment:
sequence diagram is easy to understand

Reply:
<img width="563" alt="image" src="https://user-images.githubusercontent.com/70333123/198496323-2a059ca5-e3ee-482b-8bae-1610a3bda244.png">
I think the activation bar should end after each return call here as well